clang-tools-extra/clang-tidy/readability/UseStdMinMaxCheck.cpp

```diff
@@ -0,0 +1,129 @@
+  if (!BinaryOp || If->hasElseStorage())
+    return;
+
+  if (Compound) {
+    if (Compound->size() > 1)
+      return;
+  }
+
+  const SourceLocation IfLocation = If->getIfLoc();
+  const SourceLocation ThenLocation = If->getEndLoc();
+
+  if (IfLocation.isMacroID() || ThenLocation.isMacroID())
+    return;
+
+  const auto CreateReplacement = [&](bool UseMax) {
+    const auto *FunctionName = UseMax ? "std::max" : "std::min";
+    const auto CondLhsStr = Lexer::getSourceText(
+        Source.getExpansionRange(CondLhs->getSourceRange()), Source, LO);
+    const auto CondRhsStr = Lexer::getSourceText(
+        Source.getExpansionRange(CondRhs->getSourceRange()), Source, LO);
+    const auto AssignLhsStr = Lexer::getSourceText(
+        Source.getExpansionRange(AssignLhs->getSourceRange()), Source, LO);
+    return (AssignLhsStr + " = " + FunctionName + "(" + CondLhsStr + ", " +
+            CondRhsStr + ");")
+        .str();
+  };
+  const auto OperatorStr = BinaryOp->getOpcodeStr();
+  if (((BinaryOp->getOpcode() == BO_LT || BinaryOp->getOpcode() == BO_LE) &&
+       (tidy::utils::areStatementsIdentical(CondLhs, AssignRhs, Context) &&
+        tidy::utils::areStatementsIdentical(CondRhs, AssignLhs, Context))) ||
+      ((BinaryOp->getOpcode() == BO_GT || BinaryOp->getOpcode() == BO_GE) &&
+       (tidy::utils::areStatementsIdentical(CondLhs, AssignLhs, Context) &&
+        tidy::utils::areStatementsIdentical(CondRhs, AssignRhs, Context)))) {
+    diag(IfLocation, "use `std::min` instead of `%0`")
+        << OperatorStr
+        << FixItHint::CreateReplacement(SourceRange(IfLocation, ThenLocation),
+                                        CreateReplacement(/*UseMax=*/false))
+        << IncludeInserter.createIncludeInsertion(
+               Source.getFileID(If->getBeginLoc()), AlgorithmHeader);
+
+  } else if (((BinaryOp->getOpcode() == BO_LT ||
+               BinaryOp->getOpcode() == BO_LE) &&
+              (tidy::utils::areStatementsIdentical(CondLhs, AssignLhs,
+                                                   Context) &&
+               tidy::utils::areStatementsIdentical(CondRhs, AssignRhs,
+                                                   Context))) ||
+             ((BinaryOp->getOpcode() == BO_GT ||
+               BinaryOp->getOpcode() == BO_GE) &&
+              (tidy::utils::areStatementsIdentical(CondLhs, AssignRhs,
+                                                   Context) &&
+               tidy::utils::areStatementsIdentical(CondRhs, AssignLhs,
+                                                   Context)))) {
+    diag(IfLocation, "use `std::max` instead of `%0`")
+        << OperatorStr
+        << FixItHint::CreateReplacement(SourceRange(IfLocation, ThenLocation),
+                                        CreateReplacement(/*UseMax=*/true))
+        << IncludeInserter.createIncludeInsertion(
+               Source.getFileID(If->getBeginLoc()), AlgorithmHeader);
```

Comment:
put here <algorithm> instead of AlgorithmHeader or add static const llvm::StringRef AlgorithmHeader("<algorithm>"); somewere in this .cpp (as global variable)